### PR TITLE
Plot and check classes separately for binary models. #945

### PR DESCRIPTION
**Pull request by yuyan9**
This analyzes the two classes separately for binary models rather than just checking the "average".

edit: #945 
Sorry, I didn't realize the issue number had to be in the comment.

### CODE REVIEW

File: scripts/analyze_training_metrics.py [hunk 1]
```diff
@@ -85,12 +85,12 @@ def parse_metric_file(metric_file_path: Path) -> Tuple[datetime, str, Dict[str,
...
 
-    assert file_path_parts[:5] == ["metric", "project", "bugbug", "train"]
-    model_name = file_path_parts[5]
-    assert file_path_parts[6:8] == ["per", "date"]
-    date_parts = list(map(int, file_path_parts[8:14]))
+    file_path_parts = metric_file_path.stem.split("_")
+    assert file_path_parts[:4] == ["metric", "project", "bugbug", "train"]
+    model_name = file_path_parts[4]
+    assert file_path_parts[5:7] == ["per", "date"]
+    date_parts = list(map(int, file_path_parts[7:13]))
```
Comment (marco-c) [inline scripts/analyze_training_metrics.py:93]: Could you make a separate PR with this fix? I prefer to have self-contained PRs.

File: scripts/analyze_training_metrics.py [hunk 2]
```diff
@@ -124,7 +124,16 @@ def analyze_metrics(
 
         date, model_name, metric = parse_metric_file(metric_file_path)
 
+        is_binary = len(metric["report"]["targets"]) == 2
         # Then process the report
+        if is_binary:
+            for target in range(2):
```
Comment (marco-c) [inline scripts/analyze_training_metrics.py:130]: Instead of "hardcoding" the class names like this, you could do `for target in metric["report"]["targets"]:` or even better `for target, target_metrics in metric["report"]["targets"].items():`, so in the next line you can do `for key, value in target_metrics.items():` instead of `for key, value in metric["report"]["targets"][str(target)].items():`.

**Comment by marco-c** [comment]
Thanks, looks good overall, just a couple of minor things :)

File: scripts/analyze_training_metrics.py (hunk 2 above)
Comment (yuyan9) [reply scripts/analyze_training_metrics.py:130]: Thank you for the tip!

File: scripts/analyze_training_metrics.py (hunk 1 above)
Comment (yuyan9) [reply scripts/analyze_training_metrics.py:93]: Ok done.

Comment (marco-c) [reply scripts/analyze_training_metrics.py:93]: You'll now need to rebase this PR on top of current master (`git pull upstream master --rebase`).

Comment (marco-c) [reply scripts/analyze_training_metrics.py]: It looks like it didn't go well, https://github.com/mozilla/bugbug/pull/1824/files still shows the changes from the other PR.

Comment (yuyan9) [reply scripts/analyze_training_metrics.py]: Ok, got it now I think!

**Review by marco-c — APPROVED** [review]
Thanks!
